src/tools/contextTool.ts

```diff
@@ -301,7 +395,13 @@ export function formatContextResponse(result: ContextResult): string {
     sections.push(workflowParts.join('\n'));
   }
 
-  if (sections.length === 0) {
+  // Parse handoff mode and add instructions at END for recency bias
+  const handoffMode = parseHandoffMode(result.workflow_context.content || '');
+  const handoffInstructions = getHandoffInstructions(handoffMode);
+  sections.push(`<handoff_instructions>\n${handoffInstructions}\n</handoff_instructions>`);
+
+  if (sections.length === 1 && sections[0].includes('<handoff_instructions>')) {
+    // Only handoff instructions present means no actual context was found
     return '<context status="empty" />';
```

Comment:
This failure case can return from the function early rather than checking the specific section that didn't need to be added, which seems brittle.